### DISCUSSION TITLE
Decoding errors from a secondary contract

### DISCUSSION THREAD

**Discussion by 0xJem** [wevm/viem#2026] [Question] [answered]
We have a design where a contract (`ContractA`) has modules (e.g. `ContractB`) that are installed. 
When interacting with `ContractA`, it will call `ContractB`. 
If incorrect parameters are given, `ContractB` will revert.

The call to `ContractA` from viem has the ABI of `ContractA`, so inputs, return types, errors, etc are handled fine. However, viem has no knowledge of `ContractB`, and so a revert will be displayed as an unknown error.

Is there a way to coerce viem to parse the ABI of the secondary contract? It is generally known at runtime which module (e.g. `ContractB` or `ContractC`, etc) will be used, and hence which ABI is required.

**Comment by jxom** [comment, marked as answer]
If you append the `error` ABI items from ContractB onto ContractA, then it should decode correctly.

**Reply by 0xJem** [reply]
Thanks so much!

**Reply by 0xJem** [reply]
Is there a way to do this at runtime, instead of modifying the ABI for `ContractA`?

**Reply by tmm** [reply]
You could probably create some sort of ABI dependency tree in userland. Nothing on the Viem side jumps out at the moment.
